### PR TITLE
Feature/add new templates

### PR DESCRIPTION
**Pull request by djacu**
Fixes #238. Fixes #239.

Moves previous template to a folder and adds 2 new ones.
Old template is now in `python-nixpkgs` and uses `nixpkgs` to provide packages to the kernel.
A new template, `python-flake`, uses only a flake and has no `kernels` folder.
A new template, `python-poetry`, uses `poetry` to provide packages to the kernel.

### CODE REVIEW

**Comment by MMesch** [comment]
Nice. Does this need an update to the top-level `README.md`?

File: template/python-flake/flake.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  description = "Your jupyterWith project";
+
+  nixConfig.extra-substituters = [
+    "https://tweag-jupyter.cachix.org"
+  ];
+  nixConfig.extra-trusted-public-keys = [
+    "tweag-jupyter.cachix.org-1:UtNH4Zs6hVUFpFBTLaA4ejYavPo5EFFqgd7G7FxGW9g="
+  ];
+
+  inputs.flake-compat.url = "github:edolstra/flake-compat";
```
Comment (thufschmitt) [inline template/python-flake/flake.nix:11]: What's `flake-compat` used for? Should these templates also include a `default.nix` to take advantage of it?

File: template/python-flake/flake.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  inputs.flake-compat.flake = false;
+  inputs.flake-utils.url = "github:numtide/flake-utils";
+  # TODO: inputs.nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+  inputs.nixpkgs.url = "github:nixos/nixpkgs/1158501e7c7cba26d922723cf9f70099995eb755";
```
Comment (thufschmitt) [inline template/python-flake/flake.nix:15]: Any reason why this is hardcoded?

If this is just to get a fixed known version, maybe we could ship a (partial) `flake.lock` that pins it (but doesn't prevent `nix flake update` from working)

Comment (thufschmitt) [reply template/python-flake/flake.nix:15]: Or have `inputs.nixpkgs.follows = "jupyterWith/nixpkgs"` to make sure that both stay in sync if that's desirable

**Review by thufschmitt — CHANGES_REQUESTED** [review]
A couple suggestions/questions, but looking good overall :+1: 

Definitely not a blocker here, but are these templates somehow tested (or testable)? That would be nice

**Comment by garbas** [comment]
I'm closing this PR until we finish the UX work (switching to nixos modules) then we can reopen the discussion if we want one or more templates. We will have to fix the issue of discoverability of templates, but we can discuss it once we have nixos modules in place.

**Pull request closed without merging**